lib/Mojo/Collection.pm

```diff
@@ -45,32 +45,47 @@ sub join {
   Mojo::ByteStream->new(join $_[1] // '', map {"$_"} @{$_[0]});
 }
 
-sub last { shift->[-1] }
+sub last { CORE::shift->[-1] }
 
 sub map {
-  my ($self, $cb) = (shift, shift);
+  my ($self, $cb) = (CORE::shift, CORE::shift);
   return $self->new(map { $_->$cb(@_) } @$self);
 }
 
 sub new {
-  my $class = shift;
+  my $class = CORE::shift;
   return bless [@_], ref $class || $class;
 }
 
+sub pop {
+  my $self = CORE::shift;
+  return $self->slice(0 .. ($#$self - 1));
+}
+
+sub push {
+  my $self = CORE::shift;
+  return $self->new(@$self, @_);
+}
+
 sub reduce {
-  my $self = shift;
+  my $self = CORE::shift;
   @_ = (@_, @$self);
   goto &List::Util::reduce;
 }
 
 sub reverse { $_[0]->new(reverse @{$_[0]}) }
 
+sub shift {
+  my $self = CORE::shift;
+  return $self->slice(1 .. $#$self);
```

Comment:
Doing this extra method call seems a bit slow. How about doing the array slice directly instead?